core/sql/executor/ex_exe_stmt_globals.h

```diff
@@ -307,18 +307,38 @@ class ExExeStmtGlobals : public ex_globals
 
   inline NABoolean grabMemoryQuotaIfAvailable(ULng32 size)
   { 
+    CliGlobals *cli_globals = GetCliGlobals();
+    if (cli_globals->isEspProcess())
+       return cli_globals->grabMemoryQuotaIfAvailable(size);
     if ( unusedBMOsMemoryQuota_ < size ) return FALSE;
     unusedBMOsMemoryQuota_ -= size ;
```

Comment:
I don't understand how this code relates to the code in cli/CliGlobals.cpp. There the comments suggest that the memory quota is always stored in a context object. But here we seem to be storing it in statement globals in the Master process case.

Reply:
See cli/CliGlobals.cpp line 1103